pkg/instances/multi_ig_instances_test.go

```diff
@@ -0,0 +1,161 @@
+package instances
+
+import (
+	"fmt"
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/ingress-gce/pkg/test"
+	"k8s.io/ingress-gce/pkg/utils/namer"
+	"testing"
+)
+
+const (
+	clusterID         = "clusterUID"
+	igName            = "igname"
+	zone              = "zone1"
+	emptyBasePath     = ""
+	emptyFirewallName = ""
+)
+
+func TestGetInstanceGroups(t *testing.T) {
+	recorder := &test.FakeRecorderSource{}
+	backendNamer := namer.NewNamer(clusterID, emptyFirewallName)
+	zones := []string{zone}
+	zoneLister := &FakeZoneLister{zones}
+	fakeCloud := NewFakeInstanceGroups(nil, defaultNamer)
+	multiIGInst := NewMultiIGInstances(fakeCloud, backendNamer, recorder, emptyBasePath, zoneLister)
+
+	testCases := []struct {
+		name           string
+		instanceGroups []*compute.InstanceGroup
+		expecetedIGs   int
+		expectedError  bool
+	}{
+		{
+			name:           "Single IG in cluster",
+			instanceGroups: []*compute.InstanceGroup{{Name: "k8s-ig--clusterUIDdef"}},
+			expecetedIGs:   1,
+			expectedError:  false,
+		},
+
+		{
+			name:           "Single IG, not in cluster",
+			instanceGroups: []*compute.InstanceGroup{{Name: "k8s-ig--cluster2UIDdef-1"}},
+			expecetedIGs:   0,
+			expectedError:  true,
+		},
+		{
+			name: "Multiple IGs in cluster",
+			instanceGroups: []*compute.InstanceGroup{
+				{Name: "k8s-ig--clusterUIDdef"},
+				{Name: "k8s-ig--clusterUIDdef-1"},
+				{Name: "k8s-ig--clusterUIDdef-2"},
+				{Name: "k8s-ig--clusterUIDdef-3"},
+				{Name: "k8s-ig--clusterUIDdef-5"},
+			},
+			expecetedIGs:  5,
+			expectedError: false,
+		},
+		{
+			name: "Multiple IGs, some in cluster",
+			instanceGroups: []*compute.InstanceGroup{
+				{Name: "k8s-ig--cluster1UIDdef"},
+				{Name: "k8s-ig--cluster2UIDdef-1"},
+				{Name: "k8s-ig--clusterUIDdef-2"},
+				{Name: "k8s-ig--clusterUIDdef-3"},
+				{Name: "k8s-ig--cluster1UIDdef-5"},
+			},
+			expecetedIGs:  2,
+			expectedError: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fakeCloud.instanceGroups = tc.instanceGroups
+			result, err := multiIGInst.Get(igName, "zone1")
```

Comment:
```suggestion
			result, err := multiIGInst.Get(igName, zone)
```
this is already a const